### PR TITLE
Proposal for using abstractmethods and registering virtual classes

### PR DESCRIPTION
**Pull request by JosXa**

### CODE REVIEW

File: pyrogram/client/ext/base_client.py [hunk 1]
```diff
@@ -75,15 +76,16 @@ def __init__(self):
         self.download_queue = Queue()
         self.download_workers_list = []
 
-    def send(self, data: Object):
-        pass
+    @abstractmethod
+    def send(self, data: Object): pass
```
Comment (delivrance) [inline pyrogram/client/ext/base_client.py]: Function body in the same line? Hm I don't quite like it in python, even if is just `pass`

File: pyrogram/client/methods/bots/callback_query/answer_callback_query.py [hunk 2]
```diff
@@ -20,7 +20,9 @@
 from ....ext import BaseClient
 
 
-class AnswerCallbackQuery(BaseClient):
+# noinspection PyUnresolvedReferences
```
Comment (delivrance) [inline pyrogram/client/methods/bots/callback_query/answer_callback_query.py]: I don't really like this either. How can we be 100% sure is pycharm's fault here?

File: pyrogram/client/methods/contacts/add_contacts.py [hunk 3]
```diff
@@ -32,7 +34,7 @@ def add_contacts(self, contacts: list):
             On success, the added contacts are returned.
 
         Raises:
-            :class:`Error <pyrogram.Error>`
+            :lass:`Error <pyrogram.Error>`
```
Comment (delivrance) [inline pyrogram/client/methods/contacts/add_contacts.py]: lass

File: pyrogram/client/ext/base_client.py (hunk 1 above)
Comment (JosXa) [reply pyrogram/client/ext/base_client.py]: Pretty sure that's standard for @abstractmethods

File: pyrogram/client/methods/contacts/add_contacts.py (hunk 3 above)
Comment (JosXa) [reply pyrogram/client/methods/contacts/add_contacts.py]: lol

File: pyrogram/client/methods/bots/callback_query/answer_callback_query.py (hunk 2 above)
Comment (JosXa) [reply pyrogram/client/methods/bots/callback_query/answer_callback_query.py]: Official docs use it in exactly this way. Fairly sure it's a Pycharm issue, and it works flawlessly

Comment (JosXa) [reply pyrogram/client/methods/bots/callback_query/answer_callback_query.py]: I lied. Official docs only say
"Changed in version 3.3: Returns the registered subclass, to allow usage as a class decorator."
-- https://docs.python.org/3/library/abc.html#abc.ABCMeta.register

Examples on the internet use it like this.

**Comment by delivrance** [comment]
Closing because I think abc is unnecessary boilerplate.
Change my mind.

**Pull request closed without merging**